.github/workflows/build-and-test-app.yml

```diff
@@ -0,0 +1,77 @@
+name: build-and-test
+run-name: build app from the ${{ github.head_ref }} branch
+on:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: "15 4,5 * * *"
+
+jobs:
+  build-app:
+    runs-on: macos-13
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        env: 
+          XCODE_VERSION: 15.2
+        run: sudo xcode-select -s /Applications/Xcode_${{env.XCODE_VERSION}}.app
+
+      - name: Cache Swift dependencies
+        uses: actions/cache@v4
+        with:
+         path: ~/.swiftpm
+         key: ${{ runner.os }}-swift-${{ hashFiles('**/Package.resolved') }}
+
+      - name: Install xcpretty
+        run: |
+         gem install xcpretty
+         echo 'export PATH="$(gem environment gemdir)/bin:$PATH"' >> ~/.bash_profile
```

Comment:
Does it make sense to combine build and tests into one job? If the build fails it's useless to run tests. We repeat our code without necessary. I do not see the benefits of building and testing into 2 different jobs.

Reply:
The build doesn't take much time to separate from the test job. What is the reason to spend resources on testing job?